### ISSUE TITLE
exiftool.exe hangs on Run Selection (F8)

### ISSUE THREAD

**Issue by DesertBear** [PowerShell/vscode-powershell#2559]
### System Details

System Details Output
```
### VSCode version: 1.43.0 78a4c91400152c0f27ba4d363eb56d2835f9903a x64

### VSCode extensions:       
ms-vscode.powershell@2020.3.0


### PSES version: 2.0.0.0

### PowerShell version:

Name                           Value
----                           -----
PSVersion                      7.0.0
PSEdition                      Core
GitCommitId                    7.0.0
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Issue Description

When using the Call Operator (&) and using F8 (Run Selection) to run Exiftool (the stand-alone Windows executable [HERE](https://exiftool.org/)), the terminal hangs as if it's waiting for something.  This means it's then necessary to click inside the terminal window, and then either press the 'ESC' key (which will then allow Exiftool to run), or press 'CTRL + C' (which will abort the operation).

When using the 'Terminal' menu items in VSCode, Exiftool runs OK without hanging.  

To summarise:

Exiftool runs OK when using the following methods.
Menu > Terminal > Run Active File
Menu > Terminal > Run Selected Text

Terminal hangs when using the following methods.
Menu > Run > Start Debugging (F5)
Right-clicking selected text in Editor > Run Selection (F8)

Example code (no paramenters set and assuming the Exiftool executable is saved to desktop):
`& "$env:UserProfile\Desktop\exiftool.exe"`

#### Expected Behaviour

The exiftool executable should run without hanging, like it does when using the 'Terminal' menu items.

Screenshot:
![Screenshot B](https://user-images.githubusercontent.com/27423064/76711626-9a574a80-6709-11ea-9579-185b702814b5.png)

#### Actual Behaviour

The terminal hangs without executing Exiftool, as if waiting for something, until the user intervenes by pressing either 'ESC' or 'CTRL + C' in the Terminal window.

Screenshot:
![Screenshot A](https://user-images.githubusercontent.com/27423064/76711619-90cde280-6709-11ea-908c-227962002aab.png)

**Comment by SydneyhSmith** [comment]
It would be really helpful if you could also attach the logs--instructions for how to do that are [here ](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md#providing-information-about-your-environment)

**Comment by DesertBear** [comment]
Log files attached:

[Logs.zip](https://github.com/PowerShell/vscode-powershell/files/4394474/Logs.zip)

**Comment by twest820** [comment]
+1 Not sure why this is still marked needs repro, but here's another example.

$focalLength = &"$env:USERPROFILE\Tools\Images\exiftool.exe" -FocalLength -b "$path\$fileName.MP4"

This line works fine when copy/pasted into Visual Studio Code's PowerShell terminal, when run in PowerShell 7 directly, and when run in PowerShell ISE.

Calls to other tools, such as

$resolution  = &"C:\Windows\System32\nslookup.exe" "microsoft.com"

are more successful.

### VSCode version: 1.43.2 0ba0ca52957102ca3527cf479571617f0de6ed50 x64
### VSCode extensions: ms-vscode.powershell@2020.3.0
### PSES version: 2.0.0.0
### PowerShell version:
Name                           Value
----                           -----
PSVersion                      7.0.0
PSEdition                      Core
GitCommitId                    7.0.0
OS                             Microsoft Windows 10.0.18362
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

**Comment by rjmholt** [comment]
I just managed to look into this, and got a repro on my machine specifically with `exiftool.exe`. The call operator isn't an influence here, since running the tool directly has the same effect.

I suspected this might be an issue with the integrated terminal interacting with `more`, but `more .\longfile.txt` and `help Get-Help -Full` both work as expected.

Given the specificity of this bug to a particular executable, it's unlikely we'll have time to look into this ourselves.

Given that this works in the console today, one workaround is to use VSCode's own "run in terminal" functionality, rather than the extension's (which is `$PSScriptRoot` aware). To do this with <kbd>F8</kbd>, [set this keybinding configuration](https://code.visualstudio.com/docs/getstarted/keybindings#_advanced-customization):

```json
{
 "key": "f8",
 "command": "workbench.action.terminal.runSelectedText"
 }
```

**Comment by JakeShirley** [comment]
This is an oddly specific issue that I am hitting right now, so... bump!

I've tried various things and I just cannot get this process to terminate cleanly.

**Comment by twest820** [comment]
I just checked with VS Code 1.52.1 and PowerShell 7.1 and, while two orders of magnitude slower than running in PowerShell 7 directly (or the old ISE), the VS Code console instance of pwsh.exe did see exiftool exit and get the return variable from the call populated. Unfortunately this only worked once, immediately after the update of VS Code and PowerShell, and all subsequent attempts have hung indefinitely as previously reported.

I wonder if this might perhaps be an indication VS Code is hooking into exiftool's stdout or stderr streams even though the interactions between pwsh.exe and exiftool.exe would, in principle, presumably be encapsulated to those two processed and therefore independent of code.exe. However, I can't think of an explanation for why calling exiftool would only work once after an update.

**Comment by andschwa** [comment]
@DesertBear Hi there, we just released [v2022.4.0-preview](https://github.com/PowerShell/vscode-powershell/releases/tag/v2022.4.0-preview)! Could you please try [PowerShell Preview for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-vscode.PowerShell-Preview) and verify this is fixed?

**Comment by andschwa** [comment]
This was verified resolved in https://github.com/PowerShell/vscode-powershell/issues/3225#issuecomment-1097590406
